internal/dcache/cluster_manager/cluster_manager.go

```diff
@@ -1941,9 +1942,17 @@ func (cmi *ClusterManager) updateMVList(rvMap map[string]dcache.RawVolume,
 		//
 		// Iff joinMV() is successful, consume one slot for each component RV and update existingMVMap.
 		//
-		// TODO: Set reserveBytes correctly, querying it from our in-core RV info maintained by RPC server.
+		// Get the reserveBytes correctly, querying it from our in-core RV info maintained by RPC server.
 		//
-		failedRV, err := cmi.joinMV(mvName, mv, 0 /* reserveBytes */)
+		reserveBytes, err := rm.GetMVSize(mvName)
+		if err != nil {
+			err = fmt.Errorf("failed to get disk usage of %s [%v]", mvName, err)
+			log.Err("ClusterManager::fixMV: %v", err)
+			common.Assert(false, err)
+			return
+		}
+
+		failedRV, err := cmi.joinMV(mvName, mv, reserveBytes)
```

Comment:
bette to move the reserved bytes computation inside joinMV() itself. Since we know whether we are called for a new-mv or fix-mv we can correctly compute reserved bytes inside that, simplifying the interface for the caller.

Reply:
Updated, please review